app/models/sign_up_topic.rb

```diff
@@ -0,0 +1,58 @@
+class SignUpTopic < ApplicationRecord
+  has_many :sign_up_team
+
+  validates :name, :max_choosers, presence: true
+  validates :topic_identifier, length: { maximum: 10 }
+
+  # Find if a topic is available for further selection. Topic is considered available if the
+  # total number of teams assigned to the topic are less than the maximum choosers allowed.
+  def find_if_topic_available?()
+    if @sign_up_team == nil
+      return true
+    end
+    # NOTE: Use counter_cache:true in sign_up_team to get the count of has_many relations.
+    return @sign_up_team.size < max_choosers
+  end
```

Comment:
Can this be renamed to is_available?
also the conditional here should not be to check the number of members in the @sign_up_team, it should check the number of teams(that are not waitlisted) that choose this topic should be less than max_choosers